src/deroff.rs

```diff
@@ -538,12 +538,311 @@ impl Deroffer {
+        }
+    }
+
+    fn digit(&self, _: usize) -> bool {
+        unimplemented!()
+    }
+
+    fn size(&mut self) -> bool {
+        /* # We require that the string starts with \s */
+        if self.digit(2) || ("-+".contains(Self::str_at(&self.s, 2)) && self.digit(3)) {
+            // self.skip_char(3);
+            self.s = self.skip_char(&self.s, Some(3)).to_owned();
+            while self.digit(0) {
+                // self.skip_char(1);
+                self.s = self.skip_char(&self.s, None).to_owned();
+            }
+            true
+        } else {
+            false
+        }
+    }
+
+    fn condputs(&self, _: &str) {
+        unimplemented!()
+    }
+
+    fn esc(&mut self) -> bool {
+        /* # We require that the string start with backslash */
+        if let Some(c) = self.s.get(1..2) {
+            match c {
+                "e" | "E" => self.condputs("\\"),
+                "t" => self.condputs("\t"),
+                "0" | "~" => self.condputs(" "),
+                "|" | "^" | "&" | ":" => (),
+                _ => self.condputs(c),
+            };
+            // self.skip_char(2);
+            self.s = self.skip_char(&self.s, Some(2)).to_owned();
+            true
+        } else {
+            false
+        }
+    }
+
+    fn word(&mut self) -> bool {
+        let mut got_something = false;
+        while let Some(m) = self.g_re_word.find(&self.s) {
+            got_something = true;
+            self.condputs(m.as_str());
+            // self.skip_char(m.end());
+            self.s = self.skip_char(&self.s, Some(m.end())).to_owned();
+
+            while self.spec() {
+                if !self.specletter {
+                    break;
+                }
+            }
+        }
+        got_something
+    }
+
+    fn text(&mut self) -> bool {
+        loop {
+            if let Some(idx) = self.s.find("\\") {
+                self.condputs(&self.s[..idx]);
+                // self.skip_char(idx);
+                if !self.esc_char_backslash() {
+                    // self.condputs(self.str_at(0));
+                    self.condputs(Self::str_at(&self.s, 0));
+                    // self.skip_char(1);
+                    self.s = self.skip_char(&self.s, None).to_owned();
+                }
+            } else {
+                self.condputs(&self.s);
+                self.s = String::new();
+                break;
+            }
+        }
+        true
+    }
+
+    fn spec(&self) -> bool {
+        unimplemented!()
+    }
+
+    fn esc_char_backslash(&mut self) -> bool {
+        if let Some(c) = self.s.get(1..2) {
+            match c {
+                "\"" => self.comment(),
+                "f" => self.font(),
+                "s" => self.size(),
+                "h" | "v" | "w" | "u" | "d" => self.numreq(),
+                "n" | "*" => self.var(),
+                "(" => self.spec(),
+                _ => self.esc(),
+            }
+        } else {
+            false
+        }
```

Comment:
Should take just a byte here? Can we directly just match that byte? Like
```rust
match self.s.bytes().nth(1) {
    Some('"') => self.comment(),
    ...
}
```

Reply:
`.bytes()` gets us `u8`s for each byte in the `String`, The example you have there doesn't compile. We could do `.chars()` to get the effect you're going for but I'm not a super big fan of allocating an entire iterator, going over a few elements, and then matching on an `Option` rather than just `if let`ting and then matching on a substring. I'll happily implement it your way if you'd prefer but I think it's a little misguided

Reply:
@pickfire's example would work if you adjust the match arm to be `Some(b'"')` instead of `Some('"')`.

In either case (substring vs byte), his observation of matching on `Some` rather than extracting to avoid nesting the `match` in an `if let` makes sense—no need to make `None` a special case via an `else` when `match` already handles cases like that.

Reply:
Per discussion at the meetup, the only difference in the byte vs char approach is that the `_` branch (that points to `self.esc()`) won't be triggered if we use `str::get` and the byte isn't a valid single-byte UTF-8 sequence, but _will_ be triggered if we don't require it to start and end correctly on UTF-8 boundaries. @charliethomson is doing some quick tests of the original Python to see if it can tell us one way or the other the desired behavior.

Reply:
Fixed